docs/design/operator.md

```diff
@@ -0,0 +1,414 @@
+created in different namespace that are targetting same csi driver name.
+
+```yaml
+---
+kind: CephCSIDriver
+apiVersion: csi.ceph.io/v1alpha1
+metadata:
+  name: "<prefix>.cephfs.csi.ceph.com"
+  namespace: operator-namespace
+spec:
+  deploymentName: csi-cephfsplugin-provisioner
+  daemonsetName: csi-cephfsplugin
+  Spec:
+    fsGroupPolicy: File
+    encryption:
+      configMapRef:
+        name: encryption-config-map-name
+    plugin:
+      priorityClassName: system-node-critical
+      updateStrategy:
+        type: RollingUpdate
+        rollingUpdate:
+          maxUnavailable: 1
+      labels:
+        app: cephfs-plugin
+      annotations:
+        k8s.v1.cni.cncf.io/networks: macvlan-conf-1
+    provisioner:
+      labels:
+        app: ceph-fs-provisioner
+      annotations:
+        k8s.v1.cni.cncf.io/networks: macvlan-conf-1
+      provisionerReplicas: 2
+      leaderElection:
+        leaseDuration: 100
+        renewDeadline: 100
+        retryPeriod: 10
+      attachRequired: true
+    liveness:
+      metricsPort: 8000
+    deployCSIAddons: false
+  cephfs:
+    useKernelClient: true
+    kernelMountOptions: ms_mode=secure
+status:
+  phase: Failed
+  message: Failed to create cephfs csi driver
+  reason: csi driver with same name already exists in the cluster
+```
+
+### CephCSICephConfig CRD
+
+Stores connection and configuration details to a Ceph cluster. Provides common
+connection and configuration information for multiple CSI drivers
+
+```yaml
+---
+kind: CephCSICephConfig
+apiVersion: csi.ceph.io/v1alpha1
+metadata:
+  name: ceph-cluster-1
+  namespace: operator-namespace
+spec:
+  monitors:
+  - 10.98.44.171:6789
+  - 10.98.44.172:6789
+  - 10.98.44.173:6789
+  ReadAffinity:
+    crushLocationLabels:
+    - kubernetes.io/hostname
+    - topology.kubernetes.io/region
+    - topology.kubernetes.io/zone
+  cephFS:
+    kernelMountOptions: readdir_max_bytes=1048576,norbytes
+    fuseMountOptions: debug
+  rbd:
+    mirrorDaemonCount: 2
+  config: |-
+    [global]
+    auth_cluster_required = none
+    auth_service_required = none
+    auth_client_required = none
+    rbd_validate_pool = false
+status: {}
+```
+
+### CephCSIClusterConfig CRD
+
+Contains details about CephFS subvolume group or RADOS namespace and a
+reference to CephCSICephConfig to link storage configurations to Ceph cluster.
+
+```yaml
+---
+kind: CephCSIClusterConfig
+apiVersion: csi.ceph.io/v1alpha1
+metadata:
+  name: storage
+  namespace: operator-namespace
+spec:
+  cephCSICephConfigRef:
+    name: ceph-cluster-1
+  subvolumeGroup: csi
+  radosNamespace: rados-test
+status:
+  phase: Succeeded
+  message: successfully linked to CephCSICephConnection
+```
+
+### CephCSIClusterRecovery
+
+The CephCSIClusterRecovery CR contains the local and the remote cephcluster
+configuration which will help cephcsi to identify the blockpool alias to
+consume the volume from when doing CSI operation.
+
+```yaml
+---
+kind: CephCSIClusterRecovery
```

Comment:
We should not tie this CR to the realm of Recovery. It might be useful for other use cases in the future.
```suggestion
kind: CephCSIClusterMapping
```

Reply:
i am more inclined to keep it still as Recovery as we are trying to recover from a state that cannot be changed in the current cluster am open to alternative as well. Mapping might not be the right name